### PR TITLE
Cleanup `Impl::Model`

### PR DESCRIPTION
**Pull request by ashmaroli**
cleaning up some stuff in `impl/model.rb`

### CODE REVIEW

File: lib/wab/impl/model.rb [hunk 1]
```diff
@@ -101,9 +100,8 @@ def select(format, rid, where, filter)
             }
           else
             @map.each { |ref,obj|
-              if where.eval(obj) && (filter.nil? || filter.eval(obj))
-                matches << extract_matches(format, ref, rid, obj)
-              end
+              next unless where.eval(obj) && (filter.nil? || filter.eval(obj))
+              matches << extract_matches(format, ref, rid, obj)
```
Comment (ohler55) [inline lib/wab/impl/model.rb]: Really, how does this improve the code? To me an If statement which identifies a single logical step is more clear that breaking into two separate steps.

File: lib/wab/impl/model.rb [hunk 2]
```diff
@@ -112,7 +110,7 @@ def select(format, rid, where, filter)
         result
       end
 
-      def update(obj, rid, where, filter)
+      def update(obj, _rid, where, _filter)
         updated = []
```
Comment (ohler55) [inline lib/wab/impl/model.rb]: good idea

File: lib/wab/impl/model.rb [hunk 3]
```diff
@@ -127,22 +125,19 @@ def update(obj, rid, where, filter)
...
         @lock.synchronize {
           if where.is_a?(Expr)
             @map.each { |ref,obj|
-              if where.eval(obj) && (filter.nil? || filter.eval(obj))
-                deleted << ref
-                @map.delete(ref)
-              end
+              next unless where.eval(obj) && (filter.nil? || filter.eval(obj))
+              deleted << ref
+              @map.delete(ref)
```
Comment (ohler55) [inline lib/wab/impl/model.rb]: Don't see how this improves readability or understanding.

Comment (ashmaroli) [reply lib/wab/impl/model.rb]: `if..else` inside an `each` block.. shouldn't that have an impact on readability / performance?

Comment (ohler55) [reply lib/wab/impl/model.rb]: There is no performance difference unless the guard is more expensive but I doubt there is any at all. As for readability, most people read code similar to how they talk. So as an example, which reads more smoothly and is easier to understand.

1) For each person in the room, if they are wearing a hat, smile.
2) For each person in the room, if they are not wearing a hat continue to the next person. Still here, smile.

Comment (ashmaroli) [reply lib/wab/impl/model.rb]: obviously option 1.. :smiley: 
that's a wonderful example to illustrate readability..

Comment (ohler55) [reply lib/wab/impl/model.rb]: 😄

Comment (ashmaroli) [reply lib/wab/impl/model.rb]: The example scenario can also be defined as:
  **Looking at each person wearing a hat and smiling..**
  *Look at each person's face.*
  *Next face if there's no hat on the current head*
  *Smile*

Comment (ashmaroli) [reply lib/wab/impl/model.rb]: That said, if there's no performance gain from using the `next`, I'll revert..

Comment (ohler55) [reply lib/wab/impl/model.rb]: Something like that, yes. First one is easy to follow. Not always that case if there are multiple conditions but for simple yes/no if is easier IMHO.

**Comment by ashmaroli** [comment]
I also contemplated on using `Object#tap` for some of the steps here.. but then again, not sure on its effect on performance..

**Pull request merged**